### DISCUSSION TITLE
Betriebsarten Umschaltung Vitodens 200-W B2HB-13

### DISCUSSION THREAD

**Discussion by Jonas-87** [openv/openv#603] [General]
Guten Tag zusammen,

ich habe vcontrold erfolgreich auf einem Raspberry Pi installiert. Und Betreibe diesen an meiner Vitodens 200-W B2HB-13 (Typenschild innen neben der Steuerung). Und wahrscheinlich eine Regelung Vitotronic 200 HO1B verbaut. Ich habe leider Grundsätzlich das Problem, dass ich nicht 100% weiß welches Protokoll und mit welchen Adressen ich arbeiten muss. Mein Ziel ist es am Ende die Daten aus Node-Red heraus zu lesen und ggf. zuverändern.

Ich verwende aktuell das 300er Protokoll mit der Device ID: 20CB. KW Protokoll verhält sich aber genauso.

vclient getDevType
gibt leider immer "UNKNOWN" zurück. Egal ob mit dem KW oder dem 300er Protokoll.

vclient getTempA
funktioniert.

vclient getBetriebArtM1 (0x2301)
funktioniert. Entspricht immer der Betriebsart die auch am Gerät eingestellt ist.

vclient -c "setBetriebArtM1 H+WW" (0x2301) im KW Protokoll funktioniert nicht. Liegt wahrscheinlich daran, dass man die Adresse nur lesen kann. Im 300er Protokoll gibt es diesen Befehl eh nicht.

In der Vito.xml habe ich mir ein setBetriebArt2323 für die Adresse 0x2323 erstellt.
vclient -c "setBetriebArt2323 H+WW" wird ausgeführt. Mit dem Ergebnis das im DIsplay an der Therme "externes Programm" steht und das Mond-Symbol für reduzierter Betrieb angezeigt wird.

vclient getBetriebArtM1 (0x2301) gibt anschließend
RED zurück

vclient getBetriebArt2323 (0x2323) gibt der vorher eingestellten Wert zurück (H+WW)

Nun habe ich die https://github.com/philippoo66/ViessData21/blob/master/DP_Listen_2.zip gefunden. Hier sind ja nochmal mehr Geräte aufgelistet. Leider bekomme ich nicht so richtig raus welche für meiner Steuerung passt.

Hat jemand vieleicht eine Ahnung wie ich an die passenden Adressen komme? Gibt es noch eine andere Möglichkeit die genaue Steuerung heraus zu finden oder auszulesen?

**Comment by Jonas-87** [comment]
Mit dem Tool https://github.com/philippoo66/ViessData21 bin ich ein stück weiter gekommen. Und konnte den Geräte Typen auslesen. 

Hier geht das Thema weiter... https://github.com/philippoo66/ViessData21/discussions/4

**Comment by adorobis** [comment]
Bingo! It is also the case for 20CB model (Vitodens 200-W) with the VScotHO1 controller. 0x2323 is the correct address for reading and changing the operation mode. Many thanks!
Btw, did you find which address allows setting the target storage temperature? I could only find so far the current storage temperature.
I'm also missing a few ones for checking status of DHW (storage) charging pump, storage and circulation pump. Those are available in the ViCare app but can't find them in this repo.

**Reply by Jonas-87** [reply]
hi, 
do you mean the hot water target temperature?
0x6300 ; R/W; 10 - 60 °C,

Storage charging pump;
0x6513; R only; 0=Off 1=On

Circulation pump:
0x6515; R only; 0=Off 1=On

[My list for Vitodens 200-W with Vitotronic 200 HO1B](https://www.viessmann-community.com/viessmann/attachments/viessmann/customers-heatpump-hybrid/45475/1/vitogate_200_typ_knx-Datenpunktliste.pdf)

**Reply by adorobis** [reply]
Great, many thanks for sharing! I have actually also found it on a this repo: https://github.com/philippoo66/ViessData21 it has data points for many devices, just need to browse through the files there.
